### PR TITLE
Clean-up, Merge 7.x-1.x, Fixing tests

### PR DESCRIPTION
**Pull request by Berdir**
Lots of clean-up, most tests are passing again, merged the commits from 7.x-1.x.

The dialog UI is work in progress. It's loading now but it's missing theming (it's partially see-through) and the clicking is not working, but that's already the case in the current version.

### CODE REVIEW

File: src/Controller/TokenTreeController.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\token\Controller\TokenAutocompleteController.
```
Comment (larowlan) [inline src/Controller/TokenTreeController.php]: c/p?

File: src/Controller/TokenTreeController.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+class TokenTreeController extends ControllerBase {
+
+  /**
+   * Page callback to output a token tree as an empty page.
+   */
+  function outputTree() {
+    $build['#title'] = $this->t('Available tokens');
+
+    $options = isset($_GET['options']) ? Json::decode($_GET['options']) : array();
```
Comment (larowlan) [inline src/Controller/TokenTreeController.php]: Can we use Request here?

File: src/Controller/TokenTreeController.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    // Force the dialog option to be false so we're not creating a dialog within
+    // a dialog.
+    $options['dialog'] = FALSE;
+    $build['tree']['#markup'] = _theme('token_tree', $options);
```
Comment (larowlan) [inline src/Controller/TokenTreeController.php]: We should be using #theme instead yeah?

File: token.pages.inc [hunk 4]
```diff
@@ -33,6 +73,10 @@ function theme_tree_table($variables) {
  * @ingroup themeable
  */
 function theme_token_tree($variables) {
+  if (!empty($variables['dialog'])) {
+    return theme_token_tree_link($variables);
```
Comment (larowlan) [inline token.pages.inc]: We shouldn't be hardcoding the function name right?

File: src/Controller/TokenTreeController.php (hunk 3 above)
Comment (Berdir) [inline src/Controller/TokenTreeController.php]: Woah, instant-review ;)

Copied this stuff over and only changed it minimally.. the problem with #theme is that I will have to somehow change $options and prefix everything with #, was too lazy for that...

File: token.pages.inc [hunk 5]
```diff
@@ -4,6 +4,46 @@
...
+ * Theme a link to a token tree either as a regular link or a dialog.
+ */
+function theme_token_tree_link($variables) {
+  if (empty($variables['text'])) {
+    $variables['text'] = t('Browse available tokens.');
+  }
+
+  if (!empty($variables['dialog'])) {
+    // @todo Move this somewhere it can use #attached.
+    _drupal_add_library('core/drupal.ajax');
```
Comment (larowlan) [inline token.pages.inc]: We should be using the dialog library, that hopefully will fix the css/transparent issue - it depends on ajax

File: token.pages.inc [hunk 6]
```diff
@@ -33,6 +78,10 @@ function theme_tree_table($variables) {
  * @ingroup themeable
  */
 function theme_token_tree($variables) {
+  if (!empty($variables['dialog'])) {
+    return _theme('token_tree_link', $variables);
```
Comment (larowlan) [inline token.pages.inc:82]: _theme isn't supposed to be called

**Pull request merged**